src/Workspaces/SerializationConverters.cs

```diff
@@ -54,10 +54,19 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
             var obj = JObject.Load(reader);
             var type = Type.GetType(obj["$type"].Value<string>());
-            
-            var guid = Guid.Parse(obj["Uuid"].Value<string>());
+
+            //if the id is not a guid, makes a guid based on the id of the node
+            Guid nodeId;
+            if (!Guid.TryParse((obj["Id"].Value<string>()), out nodeId))
+            {
+                nodeId = GuidUtility.Create(GuidUtility.UrlNamespace, (obj["Id"].Value<string>()));
```

Comment:
This pattern occurs multiple times, seems like it should become a helper function.